### ISSUE TITLE
UI Design: include inputs for name server, host IP, DNSSEC

### ISSUE THREAD

**Issue by brooksr4** [cisagov/getgov#893]
This ticket is an outcome of #817.

### Story

As a Domain manager **OR** CISA Analyst/SuperUser
I want to update all my DNS-related Data with the correct information
so that my website/infrastructure is correctly linked to my .gov domain

### Requirements

- [x] Implemented on or about **/domain/{nn}/nameservers** page
- [x] Facilitate the entry of alpha-numeric DNS name servers. (ex., ns1.example.com) {**UI already done**}
- [x] Facilitate the entry of IP addresses associated with name server, when NS is a subdomain of the requested domain.
- [x] Facilitate the entry of DNSSEC data records for the group of DNS entries, per the following:
> There are two types of DNSSEC records, _DS Data_ and _DNS Key Data_. 
- [ ] DS Data fields:
The DSDATA record contains a cryptographic hash of a DNSKEY record.
> **KeyTag** - A short numeric value to identify the record.
**Algorithm** - A DropDown Selector with a list of (selectable) algorithms, from a configurable file. (examples are Diffie-Hellman, RSA/SHA256, etc)
![image](https://github.com/cisagov/getgov/assets/62573986/1d8491ea-9506-4866-bf8b-9802cacf3217)

**Digest Type** - A DropDown Selector with a list of cryptographic hash algorithms, from a configurable file. (examples are SHA-1 or SHA-256
![image](https://github.com/cisagov/getgov/assets/62573986/145766c8-d673-4af1-ad5e-f3f7dc19f457)

**Digest** - A text field to hold a cryptographic hash value of the referenced DNSKEY-record.

- [x] DNSKey Data fields:
The DNSKEY record contains the public signing key.
> **Algorithm** - A Dropdown selector with a list of public key's cryptographic algorithm, from a configurable file. (example, RSA/SHA-256)
![image](https://github.com/cisagov/getgov/assets/62573986/1169b07d-38da-4924-9c36-0b31f87c3089)
**Public Key** - An alphanumeric text box for the Public Key Data
NOTE: Two other fields exist but are used for backward compatibility. These have fixed values.
**Flags** - Fixed Value of "KSK," (key-signing key) as we are only supporting second level domains. (DO WE NEED TO SUPPORT ZSK as we eventually support subdomains?)
**Protocol** - Fixed value of 3, always. DNSSEC

-  ### Constraints

DNS HOSTS:
- [ ] Enforce a minimum of 2 and maximum of 13 DNS entries.
- [x] A DNS entry which is a subdomain of the target domain **_must_** have an associated IP Address
- [x] The IP address associated with a DNS entry may be IPv4 or IPv6. (support both and detect/validate which type)
[DNSSEC ENTRIES:](https://www.cloudflare.com/learning/dns/dns-records/dnskey-ds-records/)
A Domain manager may optionally utilize DNSSEC, which 

### Validations

- [x] DNS entries must be verified to properly resolve the Domain to an IP address. (future) Documented in #1036 (later) 

### Acceptance Criteria

- [x] Domain Management page **(/domain/{nn}/nameservers)** includes fields for primary, secondary and additional host names and IPs

- [x] Fields are available for DS Data and DNS Data: Key Tag, Algorithm, Digest Type, Digest

### Additional Context

This is based on the following screenshot from the Verisign system: https://drive.google.com/file/d/1CkXnRQTcUHTD3lnfnk8VmrsMZV_4UzDy/view?usp=sharing

This affects page: **/domain/{xxx}/nameservers**, linked in [big picture here](https://miro.com/app/board/uXjVMuqbLOk=/?moveToWidget=3458764563494208651&cot=14): 

### Issue Links

🔄 Relates to: #817
Blocks: #1016 
[Figma designs](https://www.figma.com/file/aJbuDLJNNikqQObTuCNKQa/GetGov_Designs_ECS-Truss?type=design&node-id=939-54742&mode=design)

Clarifying Requirements in Slack Thread: https://cisa-corp.slack.com/archives/C05BDEA3C11/p1694524933555419

**Comment by abroddrick** [comment]
@PaulKuykendall this is the DS data we talked about today. This should be moved up in priority and dealt refined async so it can be slotted for sooner rather than later.

**Comment by h-m-f-t** [comment]
It'd be ideal to detect when a given address was (valid) v4 or v6, but in the absence of that having a user-selectable field defining the type would work.

**Comment by PaulKuykendall** [comment]
Paul - research actual fields required, and think thru entire slice.

**Comment by PaulKuykendall** [comment]
![image](https://github.com/cisagov/getgov/assets/62573986/4d2388a4-d6f0-400e-a1f4-8d6916a9a8ca)

**Comment by gabydisarli** [comment]
@PaulKuykendall Confused on this screencap of a drop down for Digest. On the Verisign screencap I see an open field for Digest, not a drop down. 
![image](https://github.com/cisagov/getgov/assets/107440934/60e37e3e-c13f-490d-a32e-9c8692c0ca48)

**Comment by gabydisarli** [comment]
added Figma design links. Figma designs are mostly done but small tweaks to copy here and there are happening in tandem with development

**Comment by gabydisarli** [comment]
Name servers page is complete and linked in associated dev ticket. There is still one small copy change pending but marking this done as it's 99% there.
